viz/pkg/healthcheck/healthcheck.go

```diff
@@ -69,22 +78,56 @@ func (hc *HealthChecker) RunChecks(observer healthcheck.CheckObserver) (bool, bo
 
 // VizCategory returns a healthcheck.Category containing checkers
 // to verify the health of viz components
-func (hc *HealthChecker) VizCategory() *healthcheck.Category {
+// fullCheck parameter will decide to run a full or a smaller set of healthchecks.
+func (hc *HealthChecker) VizCategory(fullCheck bool) *healthcheck.Category {
 	vizSelector := fmt.Sprintf("%s=%s", k8s.LinkerdExtensionLabel, VizExtensionName)
-	return healthcheck.NewCategory(LinkerdVizExtensionCheck, []healthcheck.Checker{
+
+	checks := append([]healthcheck.Checker{},
 		*healthcheck.NewChecker("linkerd-viz Namespace exists").
 			WithHintAnchor("l5d-viz-ns-exists").
 			Fatal().
 			WithCheck(func(ctx context.Context) error {
-				vizNs, err := hc.KubeAPIClient().GetNamespaceWithExtensionLabel(ctx, "viz")
-				if err != nil {
-					return err
-				}
+				if hc.vizNamespaceOverride == "" {
+					vizNs, err := hc.KubeAPIClient().GetNamespaceWithExtensionLabel(ctx, "viz")
+					if err != nil {
+						return err
+					}
 
-				hc.vizNamespace = vizNs.Name
-				hc.externalPrometheusURL = vizNs.Annotations[labels.VizExternalPrometheus]
-				return nil
+					hc.vizNamespace = vizNs.Name
+					hc.externalPrometheusURL = vizNs.Annotations[labels.VizExternalPrometheus]
+					return nil
+				} else {
```

Comment:
This else is not necessary given the everything inside the previous `if` will always return.